Merge feature/friendships v1

sebGilR opened the pull request:
In this milestone, the following was added:
- Friendships model
- Users can send friend requests both in the user list as well as in the user show page
- A User can accept or decline a friend request
- Unit and integration tests added

Commit:
Create friendship model and table

Commit:
User friendship methods created

Commit:
Add accept method to confirm friendship

Commit:
Start spec files

enelesmai committed:
added test unit for user model

Commit:
Create friendship controller with index, create and update
friendship index view file created

Commit:
Friendship request buttons added to views
Used helper method to define if friendship can be requested

enelesmai committed:
added feature test case for friendships

enelesmai committed:
added flash notices to request answer

enelesmai committed:
added fixes after rubocop

Commit:
remove unrecognized cops

Commit:
Add link to friendships in nav and create reciprocate friendship

wathigo commented on app/models/user.rb:
```diff
@@ -9,4 +9,20 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+
+  def friends
+    friendships.map { |f| f.friend if f.confirmed }
+  end
+
+  def friend?(user)
+    friends.include?(user)
+  end
+
+  def added?(user)
+    user == self ||
+      Friendship.where(user: user, friend: self).exists? ||
+      friendships.where(friend: user).exists?
+  end
 end
```
**[REQUIRED]**
User model has_many :inverted_friendships, class_name: "Friendship", foreign_key: "friend_id" ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friend.png)  from  Pathwright).

wathigo reviewed (changes requested):
### Changes Required
Great work @sebGilR  and @enelesmai  :clap: 
Let's fix some issue before we mark this as complete.

**REQUIRED**
* We have some failing tests. See this :point_down: 
![Screenshot from 2020-05-13 09-56-36](https://user-images.githubusercontent.com/44072711/81786089-e48d5a00-9507-11ea-8071-5eaca1bcb970.png)
* We can make the notification feature better. It's difficult to access friend requests from the views without manually typing the URL. Let's find a better way to achieve this.
* User timeline should contain user's posts and friends posts.
Currently, all users posts are being displayed on the timeline event if they are not friends with the current user.

**After implementing the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

Commit:
Fix tests and removed unnecessary

Commit:
Fix timeline to show only friends and own posts
implemented inverse friendships
add notification to menu with frienship request count

Commit:
Fix rubocop raised errors

sebGilR commented:
Thank you for the provided feedback. Changes requested have been implemented.

Torres-ssf reviewed (approved):
Milestone Approved :+1: 
Hi there team :hand: 
All required changes were implemented :clap: :clap: :clap: 

**[FOR THE NEXT MILESTONE]**
I notice you made only a few tests for the friendship behavior. It's best practice to create tests for the functionalities as well, creating users, creating comments...

Good luck with the next Milestone.

Happy coding :computer: 
**Sergio Torres**
